measures/ApplyUpgrade/measure.rb

```diff
@@ -543,7 +543,15 @@ def set_existing_system_as_heat_pump_backup(runner, measures, hpxml_bldg)
       heat_pump_type = args[:hvac_heat_pump_type]
       if not heat_pump_type.nil?
         heating_system = get_heating_system(hpxml_bldg)
-        if not heating_system.nil?
+
+        parameter_name = 'HVAC Heating Efficiency'
+        option_name = values[OpenStudio::toUnderscoreCase(parameter_name)]
+        options_measure_args, _errors = get_measure_args_from_option_names(lookup_csv_data, [option_name], parameter_name, lookup_file, runner)
+        resstock_arguments = options_measure_args[option_name]['ResStockArguments']
+
+        hvac_heating_system = resstock_arguments['hvac_heating_system']
```

Comment:
You might be able to just pass in the `hvac_heating_system` as the only new method argument instead. In this measure, you can retrieve it from earlier in the code. In the tests, you can get the value from the OSW.

Reply:
https://github.com/NREL/resstock/pull/1524/commits/e53ba50f7b942ec9746dc21a40ac1fae0159225a